src/pages/Cart.js

```diff
@@ -0,0 +1,121 @@
+//146
+import { useToast } from "@chakra-ui/toast";
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { useCartData } from "../Context-Reducer/CartData";
+import { useUserData } from "../Context-Reducer/UserDatacontext";
+import CartFunctions from "../utils/CartFunctions";
+
+import "../App.css";
+
+const Cart = () => {
+  const { AddToCartApiCall, RemoveFromCartApiCall, DeleteFromCartApiCall } =
+    CartFunctions();
+
+  const { user } = useUserData();
+  const [data, setData] = useState([]);
+  const { cartContext, setCartContext } = useCartData();
+  // eslint-disable-next-line no-unused-vars
+  const [loader, setLoader] = useState(true);
+  const toast = useToast();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const result = await axios(
+        "https://Terrarium-Backend.nadaafarook.repl.co/cart/" + user.id
+      );
+
+      if (result.data.success === true) {
+        setData([...result.data.cart.products]);
+        setCartContext([...result.data.cart.products]);
+        setLoader(false);
+      } else {
+        setLoader(false);
+        toast({
+          title: "Error in fetching user cart. Please try again",
+          status: "error",
+        });
+      }
+    };
+    fetchData();
+  }, [cartContext, setCartContext, toast, user.id, setLoader]);
+
+  const func = (state, value) => {
+    return {
+      totalPrice: Number(state.totalPrice) + Number(value.productId.price),
+      discountPrice:
+        state.discountPrice +
+        (value.productId.price * value.productId.discount) / 100,
+    };
+  };
+
+  const state = data.reduce(func, { totalPrice: 0, discountPrice: 0 });
+
+  return (
+    <div className="Cart">
+      <div>
+        {" "}
+        {data.length === 0 ? (
+          <div>
+            <h2 className="p-2 center">No products added in cart</h2>
+          </div>
+        ) : (
+          data.map(
+            ({
+              productId: { name, price, description, image, _id, discount },
```

Comment:
i did not understand why productId have productDetails

Reply:
I kept the ref id in productId . So after populating it is containing the details.